core-rust/core-api-server/core-api-schema.yaml

```diff
@@ -3734,6 +3734,7 @@ components:
         - GlobalValidator
         - GlobalGenericComponent
         - GlobalAccount
+        - GlobalAccountLocker
```

Comment:
hmm, should we also add it to `ALL_ENTITY_TYPES` (https://github.com/radixdlt/babylon-node/blob/f9ba2ce9ea1761d35f5365c5092bec9f9453fb53/core-rust/core-api-server/src/core_api/handlers/status_network_configuration.rs#L88), which drive `NetworkConfigurationResponse.address_types`?

AFAIS, a couple more original `EntityType` enum values are missing from there (e.g. `GlobalMultiResourcePool`)
- is this deliberate?
  - if "yes": how do we decide what is worthy of being included?
  - if "no": lemme fix this by actually iterating over the original `EntityType`.

Reply:
Not deliberate! Good catch!

I wonder if we should add a rust unit test which validates we have all `EntityType`s? This would either require adding some strum derive to `EntityType` to allow iterating over all variants, or making use of the fact `EntityType` implements SborDescribe to:
```
    let (type_id, schema) =
        generate_full_schema_from_single_type::<SimpleStruct, ScryptoCustomSchema>();
    let type_metadata = schema.into_latest().resolve_type_metadata(type_id).unwrap();
    let ChildNames::EnumVariants(variants) = type_metadata.child_names.unwrap() else {
        panic!("Not an enum");
    };
   // Compare variants with ALL_ENTITY_TYPES
```

Reply:
I don't believe we have a list of well known addresses anywhere, so we can't do the same trick there unfortunately.

Reply:
> I wonder if we should add a rust unit test which validates we have all EntityTypes?

why validate in test instead of enumerating (from that same source of truth) in runtime?

> This would either require adding some strum derive to EntityType to allow iterating over all variants, or making use of the fact EntityType implements SborDescribe to

We can also use the existence of `from_repr(u8)`, which de-facto provides an impl of `iter()`.

Reply:
> why validate in test instead of enumerating (from that same source of truth) in runtime?

Mostly because the schema stuff is quite slow and bulky, and we try not to generate types in production code as general rule.

> We can also use the existence of from_repr(u8), which de-facto provides an impl of iter().
I don't believe this gives access to variant names for correctness checks.

Reply:
_(please see what I mean in the latest commit)_

Reply:
> (please see what I mean in the latest commit)
Sweet! looks good to me :D 

Haha I didn't realise ENTITY_TYPES was just an array of `EntityType` :')

Reply:
Arguably the order has changed now, but that's probably fine for all intents and purposes...